cpp/src/ray/runtime/abstract_ray_runtime.cc

```diff
@@ -115,6 +115,35 @@ std::vector<bool> AbstractRayRuntime::Wait(const std::vector<std::string> &ids,
   return object_store_->Wait(StringIDsToObjectIDs(ids), num_objects, timeout_ms);
 }
 
+rpc::Address GetOwnershipInfo(const std::string &object_id_str) {
+  if (ConfigInternal::Instance().run_mode == RunMode::SINGLE_PROCESS) {
+    return {};
+  }
+  ray::internal::outer_object_id = object_id_str;
```

Comment:
I believe the thread-local outer object id should be set at the beginning of deserializing an object and reset at the end of deserializing an object, not in `GetOwnershipInfo`.

See https://github.com/ray-project/ray/blob/5ffa0bb01f89bdd1678f3e795a7b6a8bf0e616e2/java/runtime/src/main/java/io/ray/runtime/object/ObjectStore.java#L125-L129